### DISCUSSION TITLE
PC Requirement for yoloV8.

### DISCUSSION THREAD

**Discussion by Fahad-Siddiqui-ML** [ultralytics/ultralytics#11802] [Q&A] [answered]
I have 4 CCTV camera streams with resolution 1920 X 1080 with 25 FPS, and i want to detect vehicles and its Registration Number on all 4 camera stream using yoloV8l.pt inference for vehicle detection and Registration.pt (Custom model) for Registration number detection and insert data into database.

Ques 01: Core i7 12th Gen 3.6 GHZ 25MB Cache, 32 RAM, 512 SSD  with Nvidia RTX A2000 12GB is enough ?

**Comment by glenn-jocher** [comment, marked as answer]
Hello! Based on your requirements, your specified PC configuration with a Core i7 12th Gen CPU, 32 GB RAM, RTX A2000 12GB GPU should be quite capable of handling detection tasks on 4 streams simultaneously with YOLOv8. Both the YOLOv8l model for vehicle detection and your custom model for registration number detection should run efficiently given the GPU and overall system specs.

Just ensure that your system environment is properly optimized for running parallel streams, and consider using batch processing in your detection script to enhance performance. Here’s a small code snippet for setting up batch inference if you’re using the Python API:

```python
from ultralytics import YOLO

# Load the models
vehicle_model = YOLO('yolov8l.pt')
license_model = YOLO('Registration.pt')

# Process each stream, example for one stream
results = vehicle_model(source='stream1.mp4', batch=4)  # Modify as needed for parallel processing
```

Ensure you test the system under load to confirm real-world performance. Good luck with your setup! 🚗

**Reply by Fahad-Siddiqui-ML** [reply]
Hello @glenn-jocher ! thanks for your response.

**Reply by glenn-jocher** [reply]
@Fahad-Siddiqui-ML hello! You're welcome! If you have any more questions or need further assistance, feel free to ask. Happy coding! 😊

**Reply by seifeldin33** [reply]
If I have a system that uses multiple YOLO models to process violation images sent to the backend, how can I determine the necessary hardware specifications for each requirement? For example, if someone requests the use of 30 models with 100 cameras, how can I know the server requirements?

**Reply by glenn-jocher** [reply]
@seifeldin33 to determine server requirements for running multiple YOLO models, consider factors like the number of models, camera streams, resolution, and desired processing speed. It's best to conduct performance benchmarks on a smaller scale and scale up based on those results.
